middleware/stations.js

```diff
@@ -4,24 +4,18 @@ function unenvelope(envelope) {
   return envelope['data']
 }
 
-function fetchStations(callback) {
-
-  fetch('http://pomiary.gdmel.pl/rest/stations')
+function fetchStations() {
+  return fetch('http://pomiary.gdmel.pl/rest/stations')
     .then((res) => res.json())
-    .then((json) => {
-      // console.log(json);
+    .then((json) => unenvelope(json))
+    .then((result) => {
+      let list = []
```

Comment:
tablica zawsze powinna być deklarowana przez const -> zawsze masz zabezpieczenie, a jeśli chcesz zmieniać tablicę to robisz to niewprost, czyli funkcyjnie.